### ISSUE TITLE
AWS SQS/CloudWatch Scaler - Instance Profile On Keda Deployment

### ISSUE THREAD

**Issue by jralph** [kedacore/keda#316]
The current behaviour of the AWS SQS/CloudWatch scaler is that it looks for AWS environment variables on the deployment you are trying to scale, and if they are not found, it throws an error.

It would be good for this to not throw an error and to allow the go AWS library to attempt to load credentials in a usual way (using the default credential chain) and to search the environment variables on the keda deployment, as well as attempting to retrieve instance profiles.

When applying instance profiles to nodes (or to containers using kube2iam) this could be really handy and would allow you to scale jobs on based on sqs queues and cloudwatch metrics with the scaler only ever requiring read only credentials. 

Using the default chain for AWS also makes things more secure in the fact that no aws secrets need to be created or stored in kubernetes.

**Comment by SpyMachine** [comment]
@jralph was this issue fixed for you? I am running into the same exact issue and I'm looking at the PR now and I'm not sure this addresses the instance profile arn case. It seems like it still expects the environment variables to be set which shouldn't be necessary with instance profile arns.

**Comment by tomkerkhove** [comment]
Feel free to let us know @jralph

**Comment by SpyMachine** [comment]
I discussed this with I believe the feature author on Slack I think a month or so ago...He seemed to understand the issue and said it would require some refactoring.

The funny thing is that the underlying AWS client gets the credentials just fine but the error handling causes it to fail...my two cents is I don't even really think KEDA should have any involvement in the authentication process, let the container figure it out. But at the very least, you should be able to bypass it.

**Comment by tomkerkhove** [comment]
Well KEDA has to authenticate as well to check if the container should scale so there's no way around that

**Comment by SpyMachine** [comment]
Right. I just meant that KEDA should let the client do the work to authenticate using the credentials chain https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html and then the client can tell KEDA if it's successful or not.
